Pyrado/pyrado/algorithms/stopping_criteria/predefined_criteria.py

```diff
@@ -0,0 +1,117 @@
+#    or Technical University of Darmstadt, nor the names of its contributors may
+#    be used to endorse or promote products derived from this software without
+#    specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL FABIO MURATORE, HONDA RESEARCH INSTITUTE EUROPE GMBH,
+# OR TECHNICAL UNIVERSITY OF DARMSTADT BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from typing import Any, Callable, Optional
+
+from pyrado.algorithms.stopping_criteria.stopping_criterion import StoppingCriterion
+
+
+class AlwaysStopStoppingCriterion(StoppingCriterion):
+    """Stopping criterion that is always met."""
+
+    def __repr__(self) -> str:
+        return "AlwaysStopStoppingCriterion"
+
+    def __str__(self) -> str:
+        return "True"
+
+    def is_met(self, algo) -> bool:
+        """Returns `True`."""
+        return True
+
+
+class NeverStopStoppingCriterion(StoppingCriterion):
+    """Stopping criterion that is never met."""
+
+    def __repr__(self) -> str:
+        return "NeverStopStoppingCriterion"
+
+    def __str__(self) -> str:
+        return "False"
+
+    def is_met(self, algo) -> bool:
+        """Returns `False`."""
+        return False
+
+
+class CustomStoppingCriterion(StoppingCriterion):
+    """Custom stopping criterion that takes an arbitrary callable to evaluate."""
+
+    def __init__(self, criterion_fn: Callable[[Any], bool], name: Optional[str] = None):
+        """
+        Constructor.
+
+        :param criterion_fn: signature `[Algorithm] -> bool`; gets evaluated when `is_met` is called; allows for custom
+                             functionality, e.g. if an algorithm requires special treatment; the given algorithm is the
+                             same that was passed to the `is_met` method
+        :param name: name of the stopping criterion, used for `str(..)` and ´repr(..)`
+        """
+        self._criterion_fn = criterion_fn
+        self._name = name
+
+    def __repr__(self) -> str:
+        return f"CustomStoppingCriterion[_criterion_fn={repr(self._criterion_fn)}; name={self._name}]"
+
+    def __str__(self) -> str:
+        return "Custom" if self._name is None else self._name
+
+    def is_met(self, algo) -> bool:
+        """Invokes the criterion function that was passed to the constructor."""
```

Comment:
You can (and please do it) delete doc strings on functions inherited from a superclass *if* they have the same signature *and* they are not super special.